nilearn/_utils/data_gen.py

```diff
@@ -744,202 +751,625 @@ def basic_confounds(length, random_state=0):
     return confounds
 
 
-def create_fake_bids_dataset(base_dir='',
-                             n_sub=10,
-                             n_ses=2,
-                             tasks=['localizer', 'main'],
-                             n_runs=[1, 3],
-                             with_derivatives=True,
-                             with_confounds=True,
-                             confounds_tag="desc-confounds_timeseries",
-                             no_session=False,
-                             random_state=0):
-    """Creates a fake :term:`bids<BIDS>` dataset directory with dummy files.
+def generate_random_img(
```

Comment:
I guess that one too.

Reply:
I think that currently none of the function in that module have leading underscore:

https://github.com/nilearn/nilearn/blob/main/nilearn/_utils/data_gen.py

From what I understood anything in the `_utils` folder is considered "private" so the leading underscore does not have to be systematic: or at least that is what I inferred given that functions found in this folder have a mix of leading-underscore / no-leading-underscore names.

I am completely fine trying to homogenize this to all leading-underscore which would be nice because private / non-private status of the function would be in the filename rather than in the parent folder where its module can be found.

But if this is something that we want to do, I would rather do it in different PR and in a systematic manner.

Reply:
Ok will merge this and open an issue to homogenize function names in the _utils folder.